util/lambda_trigger/releases.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	semver "github.com/coreos/go-semver/semver"
+)
+
+// ReleaseResponse object format from releases.hashicorp.com/*/index.json
+type ReleaseResponse struct {
+	Name     string
+	Versions map[string]ReleaseVersion
+}
+
+// ReleaseVersion name, version, and shasum of a release
+type ReleaseVersion struct {
+	Name             string
+	Version          string
+	Shasums          string
+	ShasumsSignature string
+	Builds           []ReleaseBuild
+}
+
+// ReleaseBuild individual build information of a release
+type ReleaseBuild struct {
+	Name     string
+	Version  string
+	OS       string
+	Arch     string
+	Filename string
+	URL      string
+}
```

Comment:
It's probably unlikely that the format will change any time soon, but I'm wondering if we should move these structs out of `hc-releases` into a public repo, so they stay in one place and can be imported easily here and in any other projects

Reply:
Nice, yeah that was my thought as well since it will likely be reused for the linux-packaging trigger

Reply:
Discovered https://github.com/abennett/hashi-releases, going to refactor to use this instead

Reply:
Reworked!